source/ezrassor_topic_switch/__main__.py

```diff
@@ -0,0 +1,78 @@
+"""Execute a ROS node using the ezrassor_topic_switch module.
+
+This node publishes messages from one of two input topics to an output topic.
+It switches between inputs based on a mutable override status.
+
+This file contains any ROS-specific objects required to execute the module as a
+node--the rest of the module contains *no* references to ROS. This file, and
+only this file, imports rclpy for ROS API access.
+
+references:
+  docs.python.org/3/library/__main__.html
+  docs.ros2.org/foxy/api/rclpy/api.html
+  github.com/ros2/examples/tree/foxy/rclpy
+"""
+import ezrassor_topic_switch as switch
+import rclpy
+import std_msgs.msg
+import sys
+
+
+PACKAGE = "ezrassor_topic_switch"
+NODE = "topic_switch"
+PROGRAM = f"ros2 run {PACKAGE} {NODE}"
+OVERRIDE_TOPIC = "override"
+PRIMARY_INPUT_TOPIC = "primary_input"
+SECONDARY_INPUT_TOPIC = "secondary_input"
+OUTPUT_TOPIC = "output"
+QUEUE_SIZE = 10
+
+
+def main(passed_args=None):
+    """Main entry point for the ROS node."""
+    try:
+        rclpy.init(args=passed_args)
+        node = rclpy.create_node(NODE)
+
+        message_type = switch.load_message_type(sys.argv[1:], PROGRAM)
+
+        publish = node.create_publisher(
+            message_type,
+            OUTPUT_TOPIC,
+            QUEUE_SIZE,
+        ).publish
+
+        override_status = switch.OverrideStatus()
+        node.create_subscription(
+            std_msgs.msg.Bool,
+            OVERRIDE_TOPIC,
+            lambda message: override_status.update(message.data),
+            QUEUE_SIZE,
+        )
+
+        node.create_subscription(
+            message_type,
+            PRIMARY_INPUT_TOPIC,
+            switch.conditionally_process(
```

Comment:
Should we abstract what is going on here to a separate function call? I'm also not entirely sure what the function is supposed to do tbh.

Reply:
Okay, I see that it is needed for the OverrideSignal but is there another simpler way to do that than having to rely on two additional classes? Genuinely asking from a lack of understanding on my part.

Reply:
As discussed offline, Ron will investigate potential simplifications in a follow-up PR.